client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -581,16 +581,14 @@ static URI buildUri(String pathPrefix, String path, Map<String, String> params)
         Objects.requireNonNull(path, "path must not be null");
         try {
             String fullPath;
-            if (pathPrefix != null) {
-                if (path.startsWith("/")) {
-                    fullPath = pathPrefix + path;
-                } else {
-                    fullPath = pathPrefix + "/" + path;
-                }
+            if (pathPrefix != null && !pathPrefix.isEmpty()) {
+                fullPath = (pathPrefix + "/" + path).replaceAll("//+", "/");
             } else {
                 fullPath = path;
             }
 
+            logger.debug("Building URI with fullpath: ["+ fullPath+"] for path: ["+path+"]");
```

Comment:
nit: I wonder if this new log line is necessary, given that it's called for every single request I'd try to do without it. If users want to log this though they can enable request tracing?